src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallCommand.cs

```diff
@@ -216,31 +220,73 @@ public void InstallWorkloads(IEnumerable<WorkloadId> workloadIds, bool skipManif
             }
         }
 
-        private IEnumerable<string> GetPackageDownloadUrls(IEnumerable<WorkloadId> workloadIds, bool skipManifestUpdate, bool includePreview)
+        private async Task<IEnumerable<string>> GetPackageDownloadUrlsAsync(IEnumerable<WorkloadId> workloadIds, bool skipManifestUpdate, bool includePreview)
         {
             var packageUrls = new List<string>();
+            DirectoryPath? tempPath = null;
 
-            if (_workloadInstaller.GetInstallationUnit().Equals(InstallationUnit.Packs))
+            try
             {
-                var installer = _workloadInstaller.GetPackInstaller();
+                if (!skipManifestUpdate)
+                {
+                    // Add package urls to print
```

Comment:
nit: whenever you see comments for "steps", you could extract a method here and name it AddPackagesUrlsToPrint

Reply:
So is UseManifestsToResolvePacks